object-handlers.go

```diff
@@ -946,21 +946,40 @@ func (api objectAPIHandlers) CompleteMultipartUploadHandler(w http.ResponseWrite
 		completeParts = append(completeParts, part)
 	}
 	// Complete multipart upload.
+	// Send 200 OK
+	setCommonHeaders(w)
+	w.WriteHeader(http.StatusOK)
+
+	//Send whitespace character, once every 5secs, until CompleteMultipartUpload is done
+	done := make(chan bool)
+	go func() {
+		for {
+			select {
+			case <-time.After(5 * time.Second):
+				w.Write([]byte(" "))
+			case <-done:
+				return
+			}
+		}
+	}()
+
 	md5Sum, err = api.ObjectAPI.CompleteMultipartUpload(bucket, object, uploadID, completeParts)
 	if err != nil {
 		errorIf(err, "Unable to complete multipart upload.")
 		writeErrorResponse(w, r, toAPIErrorCode(err), r.URL.Path)
+		done <- true
```

Comment:
We need some sort of API level change to accommodate this an API level change would be completeMultipartUpload should return a channel and we attempt a read on that channel will unblock itself once the completeMultipartUpload is complete or error out.